### PR TITLE
fix: update pluginUsageTracker implementation

### PR DESCRIPTION
**Pull request by KaustubhKumar05**
- [x] Combine usage across device change, plugin addition/removal
- [x] Should be session/room specific to accommodate multiple rooms
- [x] Toggle methods inside HMSKrispPlugin
- [x] Send the value via `krisp.usage` event before leave

### CODE REVIEW

File: packages/hms-video-store/src/common/PluginUsageTracker.ts [hunk 1]
```diff
@@ -19,25 +19,33 @@ class PluginUsageTracker {
...
+      case 'mediaPlugin.stats': {
+        if (this.pluginLastAddedAt.has(pluginKey)) {
+          const duration =
+            event.properties.duration || (Date.now() - (this.pluginLastAddedAt.get(pluginKey) || 0)) / 1000;
+          this.pluginUsage.set(pluginKey, (this.pluginUsage.get(pluginKey) || 0) + Math.max(duration, 0) * 1000);
+          this.pluginLastAddedAt.delete(pluginKey);
+        }
       }
     }
+    return;
```
Comment (raviteja83) [inline packages/hms-video-store/src/common/PluginUsageTracker.ts:48]: doesn't seem necessary.

File: packages/hms-video-store/src/common/PluginUsageTracker.ts [hunk 2]
```diff
@@ -19,25 +19,33 @@ class PluginUsageTracker {
...
+        return;
+      }
+      case 'mediaPlugin.toggled.on':
+      case 'mediaPlugin.added': {
+        const addedAt = event.properties.added_at || Date.now();
+        this.pluginLastAddedAt.set(pluginKey, addedAt);
+        break;
+      }
+      case 'mediaPlugin.toggled.off':
+      case 'mediaPlugin.stats': {
```
Comment (raviteja83) [inline packages/hms-video-store/src/common/PluginUsageTracker.ts:39]: add a default case as well. do nothing

File: packages/hms-video-store/src/plugins/video/HMSVideoPlugin.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export interface HMSVideoPlugin {
    * variables, loading ML models etc. This can be used by a plugin to ensure it's prepared at the time
    * processVideoFrame is called.
    */
-  init(): Promise<void>;
+  init(sessionID?: string): Promise<void>;
```
Comment (raviteja83) [inline packages/hms-video-store/src/plugins/video/HMSVideoPlugin.ts:32]: we should call init with sessionID also no

File: packages/hms-video-store/src/plugins/audio/HMSAudioPluginsManager.ts [hunk 4]
```diff
@@ -87,6 +87,9 @@ export class HMSAudioPluginsManager {
     }
 
     await this.validateAndThrow(name, plugin);
+    // @ts-ignore
+    plugin.setEventBus?.(this.eventBus);
+    // @ts-ignore
```
Comment (raviteja83) [inline packages/hms-video-store/src/plugins/audio/HMSAudioPluginsManager.ts:92]: not needed

File: packages/roomkit-react/src/Prebuilt/components/AudioVideoToggle.tsx [hunk 5]
```diff
@@ -145,6 +145,10 @@ const NoiseCancellation = () => {
           onClick={e => e.stopPropagation()}
           onCheckedChange={value => {
             setActive(value);
+            if (isPluginAdded) {
+              plugin.toggle();
+              console.log(plugin, 'lggr');
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/AudioVideoToggle.tsx:150]: not needed

File: packages/hms-video-store/src/transport/index.ts [hunk 6]
```diff
@@ -495,9 +498,10 @@ export default class HMSTransport {
     this.joinParameters = undefined;
     HMSLogger.d(TAG, 'leaving in transport');
     try {
-      const sessionID = this.store.getRoom()?.sessionId || '';
-      this.eventBus.analytics.publish(AnalyticsEventFactory.getKrispUsage(sessionID));
-      this.eventBus.analytics.publish(AnalyticsEventFactory.transportLeave());
+      this.eventBus.analytics.publish(
+        AnalyticsEventFactory.getKrispUsage(this.pluginUsageTracker.getPluginUsage('HMSKrispPlugin')!),
+      );
+      this.pluginUsageTracker.updatePluginUsageData(transportLeaveEvent);
```
Comment (raviteja83) [inline packages/hms-video-store/src/transport/index.ts:504]: is this required if we are doing the above?

Comment (KaustubhKumar05) [reply packages/hms-video-store/src/transport/index.ts:504]: Currently yes, this is the cleanup. Can combine the calculation + cleanup, was initially kept separate to accommodate the singleton approach. Will remove the extra event too

**Review by raviteja83 — APPROVED** [review]

**Pull request merged**